components/raftstore-v2/src/operation/command/admin/compact_log.rs

```diff
@@ -288,19 +298,16 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
         }
     }
 
-    pub fn maybe_compact_log_from_engine<T>(
-        &mut self,
-        store_ctx: &mut StoreContext<EK, ER, T>,
-        old_index: Either<u64, u64>,
-    ) {
+    fn maybe_compact_log_from_engine<T>(&mut self, store_ctx: &mut StoreContext<EK, ER, T>) {
         let truncated = self.entry_storage().truncated_index();
-        let persisted = self.storage().apply_trace().persisted_apply_index();
-        match old_index {
-            Either::Left(old_persisted) if old_persisted >= truncated => return,
-            Either::Right(old_truncated) if old_truncated >= persisted => return,
-            _ => {}
-        }
-        let compact_index = std::cmp::min(truncated, persisted);
+        // We need to keep at lease one log at apply index.
```

Comment:
I don't understand, engine.gc(compact_index) will preserve the log at `compact_index`. Doesn't this mean this line is unneeded, and we should either set truncated_state to `compact_index - 1` or call engine gc with `compact_index + 1`?